src/lib/modules/specialconfigs/index.js

```diff
@@ -83,22 +83,33 @@ class SpecialConfigs {
     const self = this;
 
     this.embark.registerActionForEvent("contracts:deploy:afterAll", (cb) => {
-      let afterDeployCmds = self.config.contractsConfig.afterDeploy || [];
-      async.mapLimit(afterDeployCmds, 1, (cmd, nextMapCb) => {
-        async.waterfall([
-          function replaceWithAddresses(next) {
-            self.replaceWithAddresses(cmd, next);
-          },
-          self.replaceWithENSAddress.bind(self)
-        ], nextMapCb);
-      }, (err, onDeployCode) => {
-        if (err) {
-          self.logger.trace(err);
-          return cb(new Error("error running afterDeploy"));
-        }
+      if (typeof self.config.contractsConfig.afterDeploy === 'function') {
+        this.getAfterDeployLifecycleHookDependencies().then(dependencies => {
```

Comment:
Might be worth putting the whole function in `async/await` as well, so that `getAfterDeployLIfecyclehookDependencies()` can be called with `await`, meaning also that `afterDeploy()` can be called with `await` without wrapping it in an IIFE.